### ISSUE TITLE
null value in mutect2 subworkflow

### ISSUE THREAD

**Issue by maxulysse** [nf-core/sarek#1274]
### Description of the bug

> A USER ERROR has occurred: Bad input: Sample E14_null is not in BAM header: [E14_E14_1_BM_T, E14_E14_2_BM_T, E14_E14_3_LN_T, E14_E14_N]

> gatk --java-options "-Xmx29491M -XX:-UsePerfData" \ Mutect2 \ --input E14_2_BM_T.converted.cram --input E14_N.converted.cram --input E14_3_LN_T.converted.cram --input E14_1_BM_T.converted.cram \ --output E14.mutect2.vcf.gz \ --reference human_g1k_v37_decoy.fasta \ --panel-of-normals pon.vcf.gz \ --germline-resource af-only-gnomad.raw.sites.vcf.gz \ --intervals 1_65565-65573.bed \ --tmp-dir . \ --f1r2-tar-gz E14.mutect2.f1r2.tar.gz --normal-sample E14_null

> I am also hitting this error with version 3.3.2. The weird thing is that some MUTECT2_PAIRED jobs finish successfully while others fail due to null normal_id . I checked the contents of the input_intervals in the somatic_mutect2 subworkflow and the normal_id meta values are set as they are supposed to. Very confusing...

**Comment by berguner** [comment]
It looks like this happens when `--joint_mutect2` is set to `true` because the `normal_id` is removed prematurely in these lines:
https://github.com/nf-core/sarek/blob/f034b737630972e90aeae851e236f9d4292b9a4f/subworkflows/local/bam_variant_calling_somatic_mutect2/main.nf#L42-L49

@FriederikeHanssen do you remember why you removed the `normal_id` prior to running Mutect2?

**Comment by FriederikeHanssen** [comment]
probably for grouping reasons. I tried to be as conservative as possible in the end with removing stuff, looks like it wasn't good enough. I think I know how to fix it.

**Comment by FriederikeHanssen** [comment]
@asp8200  not really. We have to overwrite the normal id entry with a fixed value in the tests at the moment, so this won't bepicked up. Not overwriting it leads to other issues. Alternatively someone would need to sit down and regenerate a large chunk of the test data.

**Comment by berguner** [comment]
IIRC, we need to override the normal id because the sample names in the cram files (@SM tags) don't match the sample sheet. I considered updating them for Mutect2 tests but I wasn't sure how it would effect the other tests, so it still remains at the back of my mind...

**Comment by FriederikeHanssen** [comment]
right probably half of the checksums would be affected...

**Comment by FriederikeHanssen** [comment]
fixed in #1277
